### PR TITLE
Database generators and loader scripts complete

### PR DESCRIPTION
**Pull request by JonBeyond**
The files are:
mongo-loader.js: reads a utf-8 JSON string and posts it to the mongo database
mongod-document-loader.js: generates a JSON string of fake data and writes it to a file
package.json: added a script "npm run seed"

### CODE REVIEW

File: database-seed/mongo-loader.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+const path = require('path');
+const fs = require('fs');
+const DATABASE_LINK = process.env.DATABASE_LINK || require('../config.js').DATABASE_LINK;
```
Comment (jasonto510) [inline database-seed/mongo-loader.js]: How does the process.env work?

Comment (JonBeyond) [reply database-seed/mongo-loader.js]: The short version is `process` is a global variable for node, and `env` is the standard method for  passing down secure keys and tokens

File: database-seed/mongo-loader.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        sort: Number
+    }]
+})
+
+let Product = mongoose.model('ProductImage', productImageScheme);
+
+const seedDatabase = () => {
+    let documents = JSON.parse(retrieveData());
+    documents.forEach((document) => {
+        Product.updateOne({'productID':document.productID}, document, {upsert: true}, (err) => {
```
Comment (jasonto510) [inline database-seed/mongo-loader.js]: You should drop your database before you seed

Comment (JonBeyond) [reply database-seed/mongo-loader.js]: Done, I have updated a new promisified version.  Please take a look!

File: database-seed/mongo-loader.js [hunk 3]
```diff
@@ -19,14 +19,37 @@ const productImageScheme = mongoose.Schema({
...
+            if (err) reject(err);
+            resolve();
+        });
+    });
+}
+
+const dropDatabase = () => {
+    return new Promise((resolve, reject) => {
+        Product.deleteMany({}, (err) => {
```
Comment (jasonto510) [inline database-seed/mongo-loader.js]: Does the delete many function delete the previous database?

File: database-seed/mongo-loader.js [hunk 4]
```diff
@@ -19,14 +19,37 @@ const productImageScheme = mongoose.Schema({
 let Product = mongoose.model('ProductImage', productImageScheme);
 
 const seedDatabase = () => {
-    let documents = JSON.parse(retrieveData());
-    documents.forEach((document) => {
-        Product.updateOne({'productID':document.productID}, document, {upsert: true}, (err) => {
-            if (err) throw err;
+    dropDatabase()
```
Comment (jasonto510) [inline database-seed/mongo-loader.js]: I like how you incorporated promises! The code looks a lot cleaner without the set time out

File: database-seed/mongo-loader.js (hunk 3 above)
Comment (JonBeyond) [reply database-seed/mongo-loader.js]: It queries the database with the query (in this case `{}`), and then deletes all responses.  By querying with `{}`, it receives all results.

**Pull request merged**